dts/bindings/dma/dma.yaml

```diff
@@ -6,7 +6,7 @@
 
 properties:
     dmas:
-      required: true
+      required: false
```

Comment:
what's the thinking on marking this as `required: false`?

Reply:
dma.yaml needs to be included in clients binding.
If this is marked true, all clients need to add dmas props, while this should remain optional

Reply:
We should than just move this into base.yaml.  I assume its ok for this to be optional for st,stm32-i2s.yaml

Reply:
Moved to base.yaml.
Though mark them as required in stm32 i2s.